### DISCUSSION TITLE
Deleting items in an array does weird things

### DISCUSSION THREAD

**Discussion by DavidCide** [livewire/livewire#6828] [7. Bugs (archived)]
Deleting items in an array does weird things

As I say in the title I have an array that I use to show a list of inputs, this list has a button to add new items or remove them.

The weird things start after trying to delete more than 2 items from the array, for some reason after deleting 2 times (always 2 times), when I hit it 3 times the item I try to delete becomes null and if I hit it again it adds another item as if it was recovering what I deleted before but also null.

I have tried deleting the item in several different ways (unset, array_splice, etc) and they all have the same result.

I also noticed another weird thing, looking in the livewire V3 documentation I saw that you could call a function that runs when a [property has been modified](https://livewire.laravel.com/docs/lifecycle-hooks#update), I added the function and saw that the first 2 times I deleted it didn't go into the function, but the third time, when it starts to go wrong it does.

The code is quite simple:

**Livewire Component**
```php
use Livewire\Component;

class MappingForm extends Component
{
    public $mappings;

    public $mappingFieldTypes = ['custom', 'schema', 'constant'];

    public function mount()
    {
        if (! $this->mappings) {
            $this->mappings = [
                'a' => ['mapping' => [], 'schema' => null],
                'b' => ['mapping' => [], 'schema' => null],
            ];
        }
    }

    public function updatedMappings()
    {
        dd("the 3rd time it enters here");
    }

    public function addRow()
    {
        $this->mappings['a']['mapping'][] = ['type' => '', 'value' => ''];
        $this->mappings['b']['mapping'][] = ['type' => '', 'value' => ''];
    }

    public function deleteRow($index)
    {
        unset($this->mappings['a']['mapping'][$index]);
        unset($this->mappings['b']['mapping'][$index]);
    }

    public function render()
    {
        return view('livewire.services.mapping-form');
    }
}
```

**The view**
```blade
<div class="grid grid-cols-5 content-center p-5 border-t gap-2">

  @foreach ($mappings['a']['mapping'] as $key => $mappingA)
      <div class="col-span-2"  :key="{{$key.'a'}}">
          <div class="flex justify-end items-center gap-4">
              <x-select label="Tipo de campo" placeholder="Tipo de campo" :options="$mappingFieldTypes"
                  wire:model="mappings.a.mapping.{{ $key }}.type" />
          </div>
      </div>

      <div class="flex justify-center items-center z-30" :key="$key.'delete'">
          <x-button negative icon="trash"  wire:loading.attr="disabled" x-on:click="$wire.deleteRow({{$key}})" />
      </div>

      <div class="col-span-2" :key="{{$key.'b'}}">
          <div class="flex justify-end items-center gap-4">

              <x-select label="Tipo de campo" placeholder="Tipo de campo" :options="$mappingFieldTypes"
                  wire:model="mappings.b.mapping.{{ $key }}.type" />

          </div>
      </div>
  @endforeach

  <div class="col-span-2"></div>
  <div class="flex justify-center z-30">
      <x-button primary label="Añadir fila" icon="plus" wire:loading.attr="disabled" wire:click.prevent="addRow" />
  </div>
  <div class="col-span-2"></div>
</div>
```
I use the last versions of livewire and laravel

This problem is driving me crazy, I don't know what else to try, thanks in advance.

**Comment by diegogit03** [comment]
Some progress?

**Reply by joshhanley** [reply]
@diegogit03 what issue are you having exactly, can you share some code?

**Comment by joshhanley** [comment]
@DavidCide only just saw this due to the comment above. But if you are still having issues with it, it's because `:key` isn't valid for html elements only nested Livewire components. So instead you should be using `wire:key`.

Hope this helps!
